### PR TITLE
Address latest RPC API breaking changes

### PR DESCRIPTION
**Pull request by jstarry**
Changes:
* Added context to pubsub notification callbacks
* Updated `getSignatureStatus` return type to include slot
* Added `getSignatureStatusBatch` method

### CODE REVIEW

**Comment by jstarry** [comment]
@CriesofCarrots I decided to merge to speed up fixing the examples for v1.0.9. I think it would still be nice to get another set of eyes on these changes though when you have time

File: module.flow.js [hunk 1]
```diff
@@ -63,6 +65,11 @@ declare module '@solana/web3.js' {
     | SignatureSuccess
     | TransactionError;
 
+  declare export type SignatureStatus = {
+    slot: number,
+    status: SignatureSuccess | TransactionError,
```
Comment (CriesofCarrots) [inline module.flow.js]: `SignatureStatusResult`?

File: test/connection.test.js [hunk 2]
```diff
@@ -1011,18 +1025,64 @@ test('transaction', async () => {
...
-  await expect(connection.getSignatureStatus(signature)).resolves.toEqual({
-    Ok: null,
-  });
+
+  const response = await connection.getSignatureStatus(signature);
+  if (response !== null) {
+    expect(typeof response.slot).toEqual('number');
+    expect(response.status).toEqual({Ok: null});
+  } else {
+    expect(response).not.toBeNull();
```
Comment (CriesofCarrots) [inline test/connection.test.js]: Is this pattern right?
```
if (reponse === null) {
  expect(response).not.toBeNull();
}
```

Comment (jstarry) [reply test/connection.test.js]: Yeah, unfortunately needed to add the if statement to make flow happy

Comment (CriesofCarrots) [reply test/connection.test.js]: Oh, flow... 😕

File: module.flow.js (hunk 1 above)
Comment (jstarry) [reply module.flow.js]: Will incorporate this here when I rebase: https://github.com/solana-labs/solana-web3.js/pull/742

**Pull request merged**